ambry-cloud/src/main/java/com/github/ambry/cloud/BackupIntegrityMonitor.java

```diff
@@ -272,71 +386,33 @@ public void run() {
-      serverReplica = serverReplicationManager.createRemoteReplicaInfos(
-          Collections.singletonList(replica), store.getReplicaId()).get(0);
-      logger.info("[BackupIntegrityMonitor] Created peer server replica info [{}]", serverReplica);
-      serverScanner.addRemoteReplicaInfo(serverReplica);
-      logger.info("[BackupIntegrityMonitor] Queued peer server replica info [{}]", serverReplica);
-      long scanStartTime, scanEndTime = serverReplica.getReplicatedUntilTime();
-      long scanStopTime = System.currentTimeMillis() - SCAN_STOP_RELTIME;
-      DateFormat formatter = new SimpleDateFormat(VcrReplicationManager.DATE_FORMAT);
-      String replUntil = (String) replicaBackupProgressTokens.get(serverReplica.getReplicaId().getDataNodeId().getHostname())
-          .getProperties().get(VcrReplicationManager.REPLICATED_UNITL_UTC);
-      if (!replUntil.equals(String.valueOf(Utils.Infinite_Time))) {
-        scanStopTime = formatter.parse(replUntil).getTime();
-      }
-      logger.info("[BackupIntegrityMonitor] Scanning partition-{} from peer server replica [{}] until {} ({} ms)",
-          partition.getId(), replica, replUntil, scanStopTime);
-      while (scanEndTime < scanStopTime) {
-        scanStartTime = serverReplica.getReplicatedUntilTime();
-        serverScanner.replicate();
-        scanEndTime = serverReplica.getReplicatedUntilTime();
-        if (scanEndTime - scanStartTime > SCAN_MILESTONE) {
-          // Print progress, if a SCAN_MILESTONE's worth of data has been received from server
-          logger.info("[BackupIntegrityMonitor] Scanned {} num_server_blobs from peer server replica {} until {}, stop at {}",
-              serverScanner.getNumBlobScanned(), replica, formatter.format(scanEndTime),
-              formatter.format(scanStopTime));
-        }
-      }
-      logger.info("[BackupIntegrityMonitor] Scanned {} num_server_blobs from peer server replica {}",
-          serverScanner.getNumBlobScanned(), replica);
 
-      serverScanner.printKeysAbsentInServer(serverReplica);
 
-      ReplicationMetrics rmetrics = new ReplicationMetrics(cluster.getMetricRegistry(), Collections.emptyList());
-      logger.info("[BackupIntegrityMonitor] Verified cloud backup partition-{} against peer server replica {}"
-              + ", num_azure_blobs = {}, num_server_blobs = {}, num_mismatches = {}",
-          partition.getId(), replica, azureToken.getNumBlobs(), serverScanner.getNumBlobScanned(),
-          rmetrics.backupIntegrityError.getCount());
+      /** Compare metadata from server replicas with metadata from Azure */
+      List<RemoteReplicaInfo> serverReplicas =
+          serverReplicationManager.createRemoteReplicaInfos(replicas, store.getReplicaId());
+      for (RemoteReplicaInfo serverReplica : serverReplicas) {
+        /** Create a temporary map of all keys recovered from cloud */
+        HashMap<String, MessageInfo> azureBlobs = getAzureBlobsFromLocalStore(store);
```

Comment:
we only need to do this once, instead of for every server replica.

Reply:
Actually, I use this map to get a diff of keys present in server and cloud. This map contains keys present in azure. When replicating from server, i remove keys from this map that belong to server. At the end, i am left with keys that are in azure but not in server. So before each loop, i need to recreate this map.

Reply:
I understand that, but it's probably better to just do a shadow copy of this map for each replica then to recreate this map.
However, this is fine, it doesn't change its correctness.